src/db/mormot.db.sql.postgres.pas

```diff
@@ -666,7 +667,25 @@ procedure TSqlDBPostgresStatement.BindParams;
         ftNull:
           p^.VData := '';
         ftInt64:
-          Int64ToUtf8(p^.VInt64, RawUtf8(p^.VData));
+          begin
+            if p^.VDBType = INT4OID then
+            begin
+              fPGParamFormats[i] := 1; // binary
+              fPGParamLengths[i] := 4;
+              p^.VInt64 := htonl(p^.VInt64);
+              fPGParams[i] := @p^.VInt64;
+            end
+            else if p^.VDBType = INT8OID then
+            begin
+              fPGParamFormats[i] := 1; // binary
+              fPGParamLengths[i] := 8;
+              Int64Rec(p^.VInt64).Hi := htonl(Int64Rec(p^.VInt64).Hi);
+              Int64Rec(p^.VInt64).Lo := htonl(Int64Rec(p^.VInt64).Lo);
+              fPGParams[i] := @p^.VInt64;

```

Comment:
use bsaw64() - and note that you should have switches Hi and Lo for proper work

Reply:
done